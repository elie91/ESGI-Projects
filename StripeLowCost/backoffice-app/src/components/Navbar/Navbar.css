.navbar {
  display: flex !important;
  justify-content: space-between !important;
}

.navbar__links {
  display: flex;
}

.navbar__security {
  position: relative;
  display: flex;
  align-items: center;
}
