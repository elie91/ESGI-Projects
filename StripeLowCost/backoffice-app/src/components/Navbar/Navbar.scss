.navbar {
  display: flex !important;
  justify-content: space-between !important;

  &__links {
    display: flex;
  }

  &__security {
    position: relative;
    display: flex;
    align-items: center;
  }
}