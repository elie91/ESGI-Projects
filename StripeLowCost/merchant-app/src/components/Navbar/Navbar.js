import React from "react";
import { NavLink } from "react-router-dom";
import CartDropdown from "../Cart/CartDropdown";
import CartIcon from "../Cart/CartIcon";
import useProducts from "../../hooks/useProducts";

const Navbar = () => {

  const { selectors: productsSelectors } = useProducts();

  return (
    <nav className="navbar navbar-expand-lg navbar-light position-fixed w-100 bg-white">
      <div className='navbar__links'>
        <NavLink to='/' className='navbar-brand'>Site Marchand</NavLink>
      </div>
      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="navbar-nav ml-auto">
          <li className="nav-item">
            <NavLink className="nav-link" to="/credentials">Mes identifiants</NavLink>
          </li>
          <li className="nav-item">
            <NavLink className="nav-link" to="/orders">Mes commandes</NavLink>
          </li>
          <CartIcon/>
        </ul>
      </div>
      {!productsSelectors.getCartHidden() && <CartDropdown/>}

    </nav>
  );
};

export default React.memo(Navbar);
