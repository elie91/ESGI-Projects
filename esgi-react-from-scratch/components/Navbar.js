import Component from "../framework/Component.js";
import Zouav from "../framework/Zouav.js";

export default class Navbar extends Component {

    constructor(props, children) {
        super(props, children);
    }

    render() {
        return Zouav.createElement('nav', {'class': 'navbar navbar-light navbar-expand-lg bg-light'}, [
            Zouav.createElement('a', {'class': 'navbar-brand'}, "React Low Cost"),
            Zouav.createElement('ul', {'class': 'navbar-nav ml-auto'}, [
                Zouav.createElement('li', {'class': 'nav-item'}, [
                    Zouav.createElement('a', {'class': 'nav-link', 'href' : '#/homepage'}, 'Utilisateurs')
                ]),
                Zouav.createElement('li', {'class': 'nav-item'}, [
                    Zouav.createElement('a', {'class': 'nav-link', 'href' : '#/todolist'}, 'Todolist')
                ]),
                Zouav.createElement('li', {'class': 'nav-item'}, [
                    Zouav.createElement('a', {'class': 'nav-link', 'href' : '#/api'}, 'Star Wars')
                ])
            ]),
        ]);
    }
}
