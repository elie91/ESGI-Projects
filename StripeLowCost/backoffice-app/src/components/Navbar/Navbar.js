import React from "react";
import { Link, withRouter } from "react-router-dom";
import useUser from "../../hooks/useUser";
import './Navbar.css';
import Logo from "../../css/profil.jpeg";
import useMerchants from "../../hooks/useMerchants";

const Navbar = ({ history }) => {

    const { selectors: userSelectors, actions: userActions } = useUser();
    const { selectors: selectorMerchant } = useMerchants();

    const toggleDropdown = (event, selector) => {
        const dropdown = event.target.closest('.dropdown');
        const dropdownMenu = document.querySelector(`#${selector}`)
        dropdown.classList.toggle('show');
        dropdownMenu.classList.toggle('show');
    }

    const handleLogout = () => {
        localStorage.clear();
        userActions.logout();
    }

    const handleRemoveMerchant = () => {
        userActions.removeSelectedMerchant()
        history.replace("/")
    }

    return (
        <nav className="navbar navbar-expand navbar-light bg-white topbar mb-4 static-top shadow">

            <button id="sidebarToggleTop" className="btn btn-link d-md-none rounded-circle mr-3">
                <i className="fa fa-bars" />
            </button>

            <ul className="navbar-nav ml-auto">
                {userSelectors.getSelectedMerchant() && selectorMerchant.getMerchantsCount() > 1 && <li className="nav-item dropdown no-arrow">
                    <a className="nav-link dropdown-toggle" onClick={(e) => toggleDropdown(e, "dropdown1")} id="merchantDropdown" role="button" data-toggle="dropdown" aria-haspopup="true"
                        aria-expanded="false">
                        <span className="d-none d-lg-inline text-primary small">
                            Vue marchand actuel : {userSelectors.getSelectedMerchant().name}
                        </span>
                    </a>
                    <div className="dropdown-menu dropdown-menu-right shadow animated--grow-in" id="dropdown1" aria-labelledby="merchantDropdown">
                        <button className="dropdown-item" onClick={handleRemoveMerchant}>
                            <i className="fas fa-sign-out-alt fa-sm fa-fw mr-2 text-gray-400" />
                            Supprimer le filtre du marchand
                        </button>
                    </div>
                </li>}

                <div className="topbar-divider d-none d-sm-block" />

                <li className="nav-item dropdown no-arrow">
                    <a className="nav-link dropdown-toggle" onClick={(e) => toggleDropdown(e, "dropdown2")} id="userDropdown" role="button" data-toggle="dropdown"
                        aria-haspopup="true" aria-expanded="false">
                        <span className="mr-2 d-none d-lg-inline text-gray-600 small">
                            {userSelectors.getUser().email}
                        </span>
                        <img className="img-profile rounded-circle" src={Logo} />
                    </a>
                    <div className="dropdown-menu dropdown-menu-right shadow animated--grow-in" id="dropdown2" aria-labelledby="userDropdown">
                        <Link className="dropdown-item" to="/profile">
                            <i className="fa fa-user fa-sm fa-fw mr-2 text-gray-400" />
                            Mon profile
                        </Link>
                        <div className="dropdown-divider" />
                        <button className="dropdown-item" onClick={handleLogout}>
                            <i className="fas fa-sign-out-alt fa-sm fa-fw mr-2 text-gray-400" />
                            DÃ©connexion
                        </button>
                    </div>
                </li>

            </ul>

        </nav>
    )
}

export default withRouter(Navbar);
