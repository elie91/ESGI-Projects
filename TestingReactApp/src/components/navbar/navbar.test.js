import React from 'react';
import {shallow} from 'enzyme';

import Navbar from './navbar.component';

describe('Navbar component', () => {
  it('should render Navbar component', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });

});
