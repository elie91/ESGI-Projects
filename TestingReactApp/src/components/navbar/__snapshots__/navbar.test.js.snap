// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render Navbar component 1`] = `ShallowWrapper {}`;
